include/seastar/core/gate.hh

```diff
@@ -25,6 +25,10 @@
 #include <seastar/util/std-compat.hh>
 #include <exception>
 
+#ifdef SEASTAR_DEBUG
```

Comment:
why not just use `SEASTAR_DEBUG` ? do we expect user to use `GATE_HOLDER_DEBUG` directly?

Reply:
I'd like to reserve the opportunity to control it in some other way (configure/cmake option, !NDEBUG, etc.)

Reply:
ack.